lib/vector-config-macros/src/lib.rs

```diff
@@ -75,15 +76,23 @@ mod configurable_component;
 /// }
 /// ```
 #[proc_macro_attribute]
-pub fn configurable_component(args: TokenStream, item: TokenStream) -> TokenStream {
-    configurable_component::configurable_component_impl(args, item)
+pub fn configurable_component(attrs: TokenStream, item: TokenStream) -> TokenStream {
+    configurable_component::configurable_component_impl(attrs, item)
 }
 
-/// Generates an implementation of `Configurable` trait for the given container.
+/// Generates an implementation of the `Configurable` trait for the given container.
 ///
 /// In general, `#[configurable_component]` should be preferred as it ensures the other necessary derives/trait
 /// implementations are provided, and offers other features related to describing specific configuration types, etc.
 #[proc_macro_derive(Configurable, attributes(configurable))]
 pub fn derive_configurable(input: TokenStream) -> TokenStream {
     configurable::derive_configurable_impl(input)
 }
+
+/// Generates an implementation of the `NamedComponent` trait for the given container.
+///
+/// The only valid form of this macro is `#[component_name("something")]`.
+#[proc_macro_attribute]
+pub fn component_name(attrs: TokenStream, item: TokenStream) -> TokenStream {
+    component_name::component_name_impl(attrs, item)
+}
```

Comment:
I forgot why I wrote this as an attribute macro at first, but it's meant to be a derive macro and I plan to refactor it in the next PR where transforms are tackled.

An attribute macro is basically designed to replace itself, and the code it was attached to... while a derive macro is designed simply to add more code _after_ the thing it was attached to. You can accomplish derive-like behavior with an attribute macro, but the main difference is that if an attribute macro generates incorrect correct, the type you used it on is essentially omitted from the source... making it look like the type itself was never actually defined.

This behavior leads to a lot of garbage error messages, instead of the typically-nice error messages that procedural macros can produce, which is why I need to refactor it as a derive macro rather than leaving it as-is.